core/domain/exp_services.py

```diff
@@ -649,7 +649,12 @@ def process_changes(self, original_entity_names, changes):
                     continue
                 elif entity_name in self.deleted_entities:
                     self.changed_entities.append(entity_name)
-                    del self.property_changes[entity_name]
+                    # TODO(sll/anuzis): Refactor logic behind this method.
```

Comment:
Please replace with:

TODO(sll): This logic doesn't make sense for the following sequence of events: (a) an existing non-default entity is deleted, (b) a new default entity with the same name is created. Rewrite this method to take that case into account (or, better still, delete it altogether and use the frontend history diff functionality instead).